aws/transit-gateway-setup/variables.tf

```diff
@@ -0,0 +1,73 @@
+variable "name" {
+  description = "Name to be used on all the resources as identifier"
+  type        = string
+  default     = ""
+}
+
+variable "ram_name" {
+  description = "The name of the resource share of TGW"
+  type        = string
+  default     = ""
+}
+variable "ram_allow_external_principals" {
+  description = "Indicates whether principals outside your organization can be associated with a resource share."
+  type        = bool
+  default     = true
+}
+variable "ram_tags" {
+  description = "Additional tags for the RAM"
+  type        = map(string)
+  default     = {}
+}
+variable "tags" {
+  description = "Additional tags for the RAM"
+  type        = map(string)
+  default     = {}
+}
+variable "amazon_side_asn" {
+  description = "The Autonomous System Number (ASN) for the Amazon side of the gateway. By default the TGW is created with the current default Amazon ASN."
+  type        = string
+  default     = "64512"
+}
+
+variable "enable_auto_accept_shared_attachments" {
+  description = "Whether resource attachment requests are automatically accepted"
+  type        = string
+  default     = "enable"
+}
+
+variable "enable_default_route_table_association" {
+  description = "Whether resource attachments are automatically associated with the default association route table"
+  type        = string
+  default     = "enable"
+}
+
+variable "description" {
+  description = "Description of the EC2 Transit Gateway"
+  type        = string
+  default     = null
+}
+
+variable "enable_dns_support" {
+  description = "Should be true to enable DNS support in the TGW"
+  type        = string
+  default     = "enable"
+}
+
+variable "enable_vpn_ecmp_support" {
+  description = "Whether VPN Equal Cost Multipath Protocol support is enabled"
+  type        = bool
+  default     = false
+}
+
+variable "enable_default_route_table_propagation" {
+  description = "Whether resource attachments automatically propagate routes to the default propagation route table"
+  type        = string
+  default     = "enable"
+}
```

Comment:
and here

Reply:
Hi @angeloskyratzakos these are string types as part of TF `aws_ec2_transit_gateway`, Reference https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway#auto_accept_shared_attachments